src/pages/CreateProduct/index.js

```diff
@@ -44,8 +60,51 @@ const CreateProduct = () => {
 
   const {name, price, location, description, category} = product
 
-  const onSelect = category => {
-    setProduct({category})
+  useEffect(() => {
+    if (id) {
+      dispatch(productActions.loadProductDetail(id))
+    }
+  }, [dispatch, id])
+
+  useEffect(() => {
+    if (id) {
+      setOpen(false)
```

Comment:
no need to update this in effect, can move this logic to initial state of set open